python/pyspark/sql/types.py

```diff
@@ -57,7 +57,25 @@ def __ne__(self, other):
 
     @classmethod
     def typeName(cls):
-        return cls.__name__[:-4].lower()
+        typeTypeNameMap = {"DataType": "data",
+                           "NullType": "null",
+                           "StringType": "string",
+                           "BinaryType": "binary",
+                           "BooleanType": "boolean",
+                           "DateType": "date",
+                           "TimestampType": "timestamp",
+                           "DecimalType": "decimal",
+                           "DoubleType": "double",
+                           "FloatType": "float",
+                           "ByteType": "byte",
+                           "IntegerType": "integer",
+                           "LongType": "long",
+                           "ShortType": "short",
+                           "ArrayType": "array",
+                           "MapType": "map",
+                           "StructField": "struct",
```

Comment:
I guess it would not produce correct fields if `struct` is printed from `StructField`.

```python
>>> from pyspark.sql import Row
>>>
>>> df = spark.createDataFrame([[Row(a=1), 2]])
>>> cols = []
>>> for i in df.schema:
...     cols.append("`" + i.name + "`" + "\t" +  i.typeName())
...
>>> print ",\n".join(cols)
`_1`	struct,
`_2`	struct
```

because actual types are as below:

```python
>>> df.schema.simpleString()
'struct<_1:struct<a:bigint>,_2:bigint>'
```

How about the one as below?

```python
from pyspark.sql import Row

df = spark.createDataFrame([[Row(a=1), 2]])
cols = []
for i in df.schema:
    cols.append("`" + i.name + "`" + "\t" +  i.dataType.simpleString())

print ",\n".join(cols)
```

prints

```
`_1`	struct<a:bigint>,
`_2`	bigint
```